src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,249 @@
+package duke;
+
+import java.util.Scanner;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.io.IOException;
+
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Duke {
+    private static List<Task> tasks;
+
+    public static void main(String[] args) {
+        greet();
+        Scanner sc = new Scanner(System.in);
+        tasks = loadTasks();
+        String input = sc.nextLine();
+        while (true) {
+            try {
+                String[] parsedInput = input.strip().split("\\s+", 2);
+                switch (parsedInput[0]) {
+                case "bye":
+                    if (parsedInput.length > 1) {
+                        throw new InvalidCommandException();
+                    }
+                    exit();
+                    break;
+                case "list":
+                    if (parsedInput.length > 1) {
+                        throw new InvalidCommandException();
+                    }
+                    list();
+                    break;
+                case "done":
+                    try {
+                        int taskNumber = Integer.parseInt(parsedInput[1]);
+                        completeTask(taskNumber);
+                    } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
Maybe you could explore catching all errors at the end, so that you can avoid multiples try & catch in between.